lib/ansible/modules/network/infinity/infinity.py

```diff
@@ -0,0 +1,650 @@
+        matched_network_id = ""
+        resource_url = "networks/" + str(network_id) + "/children"
+        response = self._get_api_call_ansible_handler(method, resource_url)
+        if not response:
+            self.module.exit_json(
+                msg=" there is  error  in release network %r  from network  %s." %
+                (network_id, released_network_name))
+        if response:
+            response = json.loads(response)
+            for child_net in response:
+                if child_net['network'] and child_net['network']['network_name'] == released_network_name:
+                    matched_network_id = child_net['network']['network_id']
+                    break
+        response = None
+        if matched_network_id:
+            method = 'delete'
+            resource_url = "networks/" + str(matched_network_id)
+            response = self._get_api_call_ansible_handler(
+                method, resource_url, stat_codes=[204])
+        else:
+            self.module.exit_json(
+                msg=" When release network , could not find the network   %r from the given superent %r' " %
+                (released_network_name, network_id))
+
+        return response
+
+    # ---------------------------------------------------------------------------
+    # add_network()
+    # ---------------------------------------------------------------------------
+    def add_network(
+            self,
+            network_name="",
+            network_address="",
+            network_size="",
+            network_family='4',
+            network_type='lan',
+            network_location=-1):
+        """
+        add  new nework into    Infinity via rest api  into given supernet or default supernet
+        required fields=['network_name', 'network_family', 'network_type',  'network_address','network_size' ]
+
+        """
+        method = 'post'
+        resource_url = 'networks'
+        response = None
+        if network_name is None or network_address is None or network_size is None:
+            self.module.exit_json(
+                msg="You must specify  those the  options 'network_name', 'network_address' and 'network_size'")
+
+        if not network_family:
+            network_family = '4'
+        if not network_type:
+            network_type = 'lan'
+        if not network_location:
+            network_location = -1
+        payload_data = {
+            "network_name": network_name,
+            'network_address': network_address,
+            'network_size': network_size,
+            'network_family': network_family,
+            'network_type': network_type,
+            'network_location': network_location}
+        response = self._get_api_call_ansible_handler(
+            method='post',
+            resource_url=resource_url,
+            stat_codes=[200],
+            payload_data=payload_data)
+        return response
+
+
+def main():
+
+    fields = {
```

Comment:
This seems a lot longer and less readable that the standard way of defining the `argument_spec`.
https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/developing_modules_general.rst#main-and-ansiblemodule-argument-spec